test/unit/toggle/toggle-component.js

```diff
@@ -0,0 +1,151 @@
+import Toggle from '../../../src/components/toggle';
+import ToggleView from '../../../src/views/toggle';
+
+const fakeClient = {
+  fetchCart: () => { return; },
+  createCart: () => { return; },
+};
+
+describe('Toggle Component class', () => {
+  describe('constructor', () => {
+    let toggle;
+    let cart;
+
+    beforeEach(() => {
+      cart = {
+        node: document.createElement('div'),
+      };
+      document.body.appendChild(cart.node);
+    });
+
+    afterEach(() => {
+      document.body.removeChild(cart.node);
+    });
+
+    it('sets typeKey to toggle', () => {
+      toggle = new Toggle({}, {client: fakeClient, cart});
```

Comment:
Since we're not passing in a node or stubbing out the document methods, I think this test might be leaving a residual div in the DOM